### ISSUE TITLE
BufferQueue has been abandoned

### ISSUE THREAD

**Issue by inambutt487** [surendramaran/YOLOv8-TfLite-Object-Detector#14]
Hi @surendramaran  I hope you are fine. First of all, your hard work and contribution are appreciated within this repo.

I am facing an issue in which I need your guidance.  When the picture is captured and Camera Activity closes, Camera Activity closing or reopening again the app crashes, or the camera's live detection doesn't work when we launch Camera Activity second time.

you can download  model and test images here: https://github.com/inambutt487/Bottle/tree/master/app/src/main/assets
Here is text file for error log when crash occur 
[bottle detections error.txt](https://github.com/surendramaran/YOLOv8-TfLite-Object-Detector/files/15129159/bottle.detections.error.txt)

#2(BLAST Consumer)2](id:141e00000006,api:4,p:1522,c:5150) queueBuffer: BufferQueue has been abandoned

![aaa](https://github.com/surendramaran/YOLOv8-TfLite-Object-Detector/assets/31442367/ad6951d0-6f50-4130-af18-3eb40ca83527)

Issue produced: 
Step 1: Create an activity before launching MainActivity.
Step 2: Launch MainActivity on the button from the newly created activity to capture images of object detections and close the activity.
Step 3: Reopen the MainActivity from the newly created activity.

**Comment by surendramaran** [comment]
Are you using camera2?

It seems like the buffer (image) supposed to have a surface view but it couldn't get.

**Comment by 10Johannes** [comment]
Same, problem. I have a different main activity that has a button. When I click that button I run the detector, but it can't detect anything. When I stopped the application, it shows BufferQueue error.

**Comment by surendramaran** [comment]
How can I reproduce this error, can you send sample code ?

**Comment by 10Johannes** [comment]
You can try to run the MainActivity.kt on a different .kt file. Like how @inambutt487 did it, he created a TestActivity.kt to run the MainActivity.kt using the:

        binding.start.setOnClickListener {
            startActivity(Intent(this@TestActivity, MainActivity::class.java))
        }

**Comment by inambutt487** [comment]
Yes I am using the update for CameraX the same libraries that Up is using The problem within this repo is that when we launch the second camera throw intent, it does not detect or throw an exception.  If you launch the camera with a throw intent, it doesn't detect or throw an exception. You can download repo https://github.com/inambutt487/Bottle

**Comment by inambutt487** [comment]
There is an issue with tensor flow that I have noticed, and then it says that we do load the so file the first time, but the second time it does.

**Comment by surendramaran** [comment]
@inambutt487 Using a different fragment can help in "BufferQueue has been abandoned".

And about no detection:
I tried to create a new fragment for camera and I thought maybe the detection will show but it didn't. 
I have no idea how but later it started working, I don't remember what exactly I changed.

Here is the code, can you also try.

[Bottle-master 2.zip](https://github.com/user-attachments/files/15933308/Bottle-master.2.zip)


![clideo_editor_6c0e948d309d4145a16ad8d34e93ffe3-ezgif com-crop](https://github.com/surendramaran/YOLOv8-TfLite-Object-Detector/assets/65958897/7305731e-308e-4437-80a8-64728ac27ddb)

**Comment by ashutosh3ojha2007** [comment]
Hi @surendramaran I hope you are doing good. How can i add some more labels( cctv, spy camera, mobile camera, normal camera) in labels file which will detect by given your code. 

Can you guide me to generate new .tflite file ?

Thankyou.
